### PR TITLE
Added concurrent parsing feature for genbank files

### PR DESCRIPTION
**Pull request by Koeng101**
This adds simple concurrent parsing for genbank files. I added two functions:

1. ParseConcurrent: Parses a mutli-genbank file into a channel
2. ParseFlatConcurrent: Parses a flat-genbank file into a channel

I also added an MD5 hash into the Sequence struct. This will be used in allbase to see if genbank files should be updated.

### CODE REVIEW

File: io/genbank/genbank.go [hunk 1]
```diff
@@ -35,6 +38,9 @@ func Parse(file []byte) poly.Sequence {
 	// Create sequence struct
 	sequence := poly.Sequence{}
 
+	// Add the SHA256 hash to sequence.SHA256
```
Comment (TimothyStiles) [inline io/genbank/genbank.go:41]: MD5 isn't SHA256

File: poly.go [hunk 2]
```diff
@@ -103,6 +103,7 @@ type Sequence struct {
 	SequenceHashFunction string    `json:"hash_function"`
 	Sequence             string    `json:"sequence"`
 	Features             []Feature `json:"features"`
+	MD5                  [16]byte  `json:"hash"` //SHA256 checksum
```
Comment (TimothyStiles) [inline poly.go:106]: MD5 is not SHA256

File: io/genbank/genbank.go [hunk 3]
```diff
@@ -760,41 +766,29 @@ Genbank Flat specific IO related things begin here.
...
-	// and gets rid of the data in the header, which may be useful for some
-	// application. Header data is not needed to parse the Genbank files, though
-	gbkWithoutHeader := []byte(strings.Join(strings.Split(gbk, "\n")[10:], "\n"))
-
-	// Pass gbkWithoutHeader to ParseMulti, which should handle
-	// the rest of the parsing just fine
-	sequences := ParseMulti(gbkWithoutHeader)
-	return sequences
+	r := bytes.NewReader(file)
+	sequences := make(chan poly.Sequence, 1000) // A buffer is used so that the functions run as it is appending
```
Comment (TimothyStiles) [inline io/genbank/genbank.go:785]: Why 1000 channels?

File: io/genbank/genbank.go [hunk 4]
```diff
@@ -760,41 +766,29 @@ Genbank Flat specific IO related things begin here.
 
 // ParseMulti parses multiple Genbank files in a byte array to multiple sequences
 func ParseMulti(file []byte) []poly.Sequence {
+	r := bytes.NewReader(file)
+	sequences := make(chan poly.Sequence, 1000) // A buffer is used so that the functions run as it is appending
```
Comment (TimothyStiles) [inline io/genbank/genbank.go:770]: Why 1000 channels?

File: io/genbank/genbank.go [hunk 5]
```diff
@@ -35,6 +38,9 @@ func Parse(file []byte) poly.Sequence {
 	// Create sequence struct
 	sequence := poly.Sequence{}
 
+	// Add the SHA256 hash to sequence.SHA256
+	sequence.MD5 = md5.Sum(file)
```
Comment (TimothyStiles) [inline io/genbank/genbank.go:42]: Does Genbank publish their file hashes and how do we check against them?

Comment (Koeng101) [reply io/genbank/genbank.go:42]: No. This is for if you want to check if a genbank file has changed.

File: poly.go [hunk 6]
```diff
@@ -103,6 +103,7 @@ type Sequence struct {
 	SequenceHashFunction string    `json:"hash_function"`
 	Sequence             string    `json:"sequence"`
 	Features             []Feature `json:"features"`
+	MD5                  [16]byte  `json:"hash"` //MD5 checksum
```
Comment (TimothyStiles) [inline poly.go:106]: Considering that we already have sequence_hash and hash_function can we call this something like `GenbankCheckSum` in both the struct and json output @Koeng101?

**Comment by Koeng101** [comment]
Resolved the naming issue (changed from MD5 to CheckSum). Now uses blake3. Additionally added this as a feature into the gff parser. 

Interestingly enough, GFF actually produces the exact file (as hashed) as its input, meaning it is far more reliable than Genbank.

**Review by TimothyStiles — APPROVED** [review]
Looks good to me. Approved for merge.

**Pull request merged**